src/Orchard.Web/Modules/Orchard.OpenId/Constants/General.cs

```diff
@@ -1,7 +1,7 @@
 ﻿namespace Orchard.OpenId.Constants {
     public class General {
         public const string AuthenticationErrorUrl = "~/Authentication/Error";
-        public const string LogonCallbackUrl = "~/Users/Account/LogonCallback";
+        public const string LogonCallbackUrl = "/Users/Account/LogonCallback";
```

Comment:
No to be honest, I haven't test it on multi-tenancy. I will test it and update the PR if required.